jscomp/frontend/bs_builtin_ppx.ml

```diff
@@ -532,6 +545,42 @@ let rec structure_mapper ~await_context (self : mapper) (stru : Ast_structure.t)
             };
       }
       :: structure_mapper ~await_context self rest
+    | Pstr_value (_, vbs) ->
+      let item = self.structure_item self item in
+      let module_exprs =
+        vbs
+        |> List.filter_map (fun ({pvb_expr} : Parsetree.value_binding) ->
+               match pvb_expr.pexp_desc with
+               | Pexp_letmodule (_, ({pmod_attributes} as me), _)
+                 when Res_parsetree_viewer.hasAwaitAttribute pmod_attributes ->
+                 Some me
+               | _ -> None)
+      in
+      (* [ module __Belt_List__ = module type of Belt.List ] *)
+      let module_type_decls =
+        module_exprs
+        |> List.filter_map (fun ({pmod_desc} as me : Parsetree.module_expr) ->
```

Comment:
nit: instead of filter-map twice and compose, why not compose the functions inside directly, and do only 1 filter-map.

Reply:
https://github.com/rescript-lang/rescript-compiler/pull/6310/commits/7d4b94a0d687fe6e12dba4d6e1120e862634d3f4 https://github.com/rescript-lang/rescript-compiler/pull/6310/commits/1043b3528baa8d39aa79d33cf6a0ba8011b0a1c1